parts/k8s/kubernetescustomscript.sh

```diff
@@ -51,6 +52,10 @@ else
     REBOOTREQUIRED=false
 fi
 
+function testOutboundConnection() {
+    retrycmd_if_failure 20 5 20 curl www.bing.com || retrycmd_if_failure 20 5 20 curl www.ubuntu.com || exit $ERR_OUTBOUND_CONN_FAIL
```

Comment:
I would suggest, to test simply "I have outbound internet access", we should do a port 53 test against a highly available DNS resolver (e.g., google public dns: `8.8.8.8`/`8.8.4.4`, or one the root name servers, see a current root hints file here: http://www.internic.net/domain/named.root)

@khenidak @slack do you agree?

Also, are we in agreement that "we have outbound internet access" == "a single successful connection response within an interval of time"?

Reply:
> Also, are we in agreement that "we have outbound internet access" == "a single successful connection response within an interval of time"?

This test is meant to quickly error out in the case of VNETs with firewalls / NSGs that disallow outbound connection (100% of the time) so yes in that case I would say "a single successful connection" is enough to determine that we are in that situation and we want to bail.

This test is not designed to catch transient network flakiness (for now), that would need to be checked differently.